### PR TITLE
222 l build uk degree details page

### PR DESCRIPTION
**Pull request by defong**
### Context
uk degree details page

### Changes proposed in this pull request
Add uk degree details

### Guidance to review

```
Given a trainee exists
and I visit the degree page
and I select the uk degree type
and I select an uk degree
and I click continue button
and I am on the degree details page
and I filling the degree details
When I click continue button
Then I am on the summary page
```

NB: ned to check database that the degree is saved

##### Empty item preselected for uk degree type
image

##### Error summary deep link
image

##### On landing
image

##### With all validations triggered
image

### CODE REVIEW

**Comment by defong** [comment]
[[L] Build "UK Degree Details" Page](https://trello.com/c/Ykzvs4Cd/222-l-build-uk-degree-details-page)

File: app/views/trainees/degrees/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      <%- degree_subjects = DEGREE_SUBJECTS.map do |name| 
+            OpenStruct.new(name: name) 
+          end 
+          degree_subjects.unshift(OpenStruct.new(name: nil))
+      %>
+      <%= f.govuk_collection_select :degree_subject, degree_subjects, 
+            :name, :name, label: { text: 'Degree subject', size: 's' },
+            hint: { text: 'Search for the closest matching subject' } %>
+
+      <%- institutions = INSTITUTIONS.map do |name|
```
Comment (darokel) [inline app/views/trainees/degrees/edit.html.erb:15]: I feel like this should be handled in a helper?

File: app/views/trainees/degrees/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds">
+    <%= form_with model: [ @trainee, @degree ], local: true do |f| %>
+      <h2 class="govuk-heading-l"> Degree Details </h2>
+      <%= f.govuk_error_summary %>
+      <%- degree_subjects = DEGREE_SUBJECTS.map do |name|
```
Comment (darokel) [inline app/views/trainees/degrees/edit.html.erb:6]: same for this one, should it be in a helper to keep the view dumb?

File: app/views/trainees/degrees/new.html.erb [hunk 3]
```diff
@@ -10,36 +10,39 @@
           </p>
 
           <%- uk_degrees = HESA_DEGREE_TYPES.map do |_hesa_code, _abbreviation, name, _level|
```
Comment (darokel) [inline app/views/trainees/degrees/new.html.erb:12]: this one too, move to a helper?

**Review by darokel — APPROVED** [review]
Overall this looks fine. Discussed suggested changes offline to be done in a separate PR. 

Also this is minor but would be good if we update these feature spec method names so they make a little more sense:

```
then_i_see_error_summary => then_i_see_the_error_summary
and_i_filling_the_degree_details => and_i_fill_the_degree_details
and_i_click_continue_button => and_i_click_the_continue_button
```

File: app/views/trainees/degrees/edit.html.erb (hunk 1 above)
Comment (defong) [reply app/views/trainees/degrees/edit.html.erb:15]: next pr @JonF1

File: app/views/trainees/degrees/edit.html.erb (hunk 2 above)
Comment (defong) [reply app/views/trainees/degrees/edit.html.erb:6]: next pr @JonF1

File: app/views/trainees/degrees/new.html.erb (hunk 3 above)
Comment (defong) [reply app/views/trainees/degrees/new.html.erb:12]: next pr @JonF1

**Comment by defong** [comment]
> Overall this looks fine. Discussed suggested changes offline to be done in a separate PR.
> 
> Also this is minor but would be good if we update these feature spec method names so they make a little more sense:
> 
> ```
> then_i_see_error_summary => then_i_see_the_error_summary
> and_i_filling_the_degree_details => and_i_fill_the_degree_details
> and_i_click_continue_button => and_i_click_the_continue_button
> ```

Done


https://github.com/DFE-Digital/register-trainee-teachers/pull/96/commits/7b79f26150e7503599dc7c3a5885ea0fddf4ece7

File: app/views/trainees/degrees/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,43 @@
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds">
+    <%= form_with model: [ @trainee, @degree ], local: true do |f| %>
+      <h2 class="govuk-heading-l"> Degree Details </h2>
```
Comment (aliuk2012) [inline app/views/trainees/degrees/edit.html.erb:4]:
```suggestion
      <h2 class="govuk-heading-l">Degree Details</h2>
```

File: config/locales/en.yml [hunk 5]
```diff
@@ -28,6 +28,15 @@ en:
               blank: "You must select a Non-UK degree"
             locale_code:
               blank: "You must select either a UK degree or Non-UK degree"
+            degree_subject:
+              blank: "You must select a degree_subject"
```
Comment (aliuk2012) [inline config/locales/en.yml:32]:
```suggestion
              blank: "You must select a subject"
```

File: config/locales/en.yml [hunk 6]
```diff
@@ -28,6 +28,15 @@ en:
...
+            institution:
+              blank: "You must select a institution"
+            degree_grade:
+              blank: "You must select a degree grade"
```
Comment (aliuk2012) [inline config/locales/en.yml:36]:
```suggestion
              blank: "You must select a grade"
```

**Review by aliuk2012 — APPROVED** [review]
Only merge after tidying up the commits ;)

**Pull request merged**